src/SettingsManager.js

```diff
@@ -0,0 +1,118 @@
+  'applicationId',
+  'masterKey',
+  'serverURL',
+  'collectionPrefix',
+  'filesController',
+  'pushController',
+  'loggerController',
+  'hooksController',
+  'userController',
+  'authDataManager',
+  'liveQueryController',
+  'enableConfigChanges',
+  'settingsInitialized'
+];
+
+// settings defined in server configuration
+let definedSettings = {};
+
+// callbacks for specific setting changes
+let authChange = settings => settings.authDataManager = authDataManager(settings.ouath, settings.enableAnonymousUsers);
+let onChange = {
+  logLevel: settings => configureLogger({ level: settings.logLevel }),
+  verbose: settings => {
+    settings.logLevel = 'silly';
+    configureLogger({ level: settings.logLevel });
+  },
+  oauth: authChange,
+  enableAnonymousUsers: authChange,
+  sessionLength: settings => settings.sessionLength = Number(settings.sessionLength),
+}
+
+export default function SettingsManager(appId) {
+  return {
+    pull: _ => {
+      return getSettingsCollection(appId)
+        .then(coll => coll.find({ applicationId: appId }, { limit: 1 }))
+        .then(results => {
+          let settings = results && results.length && results[0];
+          if (settings) delete settings._id;
+          logger.verbose('Pulled settings: ' + JSON.stringify(settings, null, 2));
+          return settings;
+        });
+    },
+
+    push: settings => {
+      settings = deepcopy(settings);
+      settings.applicationId = appId;
+      return getSettingsCollection(appId)
+        .then(coll => coll.upsertOne({applicationId: appId}, { $set: settings }))
+        .then(_ => {
+          logger.verbose('Pushed settings: ' + JSON.stringify(settings, null, 2));
+        });
+    },
+
+    updateCache: (updates = {}) => {
+      updates = Object.keys(updates)
+        .filter(update => !equal(updates[update], cache.apps.get(appId)[update]) && !isLocked(update) && !isDefined(update))
+        .reduce((filtered, update) => {
+          filtered[update] = updates[update];
+          return filtered;
+        }, {});
+
+      Object.keys(updates).forEach(setting => {
+        var config = cache.apps.get(appId);
+        logger.info(`Setting '${setting}' updated from '${config[setting]}' to '${updates[setting]}'`);
+        config[setting] = updates[setting];
+        if (onChange[setting]) onChange[setting](cache.apps.get(appId))
+      });
+      return updates;
+    },
+
+    getUnlocked: _ => {
+      let settings = cache.apps.get(appId);
+
+      let settingsString = JSON.stringify(settings, (k, v) => {
+        if (!lockedSettings.includes(k)) {
+          if (v === undefined) return null;
+          return v;
+        }
+      });
+
+      return JSON.parse(settingsString);
+    },
+
+    setDefined: settings => definedSettings = settings
+  };
+
+  function getSettingsCollection() {
+    let config = cache.apps.get(appId);
+    return database.getDatabaseConnection(appId, config.collectionPrefix).adaptiveCollection(settingsCollectionName);
```

Comment:
Can you give me a code snippet?  Not sure how I can get access to the current dbController without the getDatabaseConnection code.